plugins/ssl/certificate_file_expiry

```diff
@@ -72,23 +87,70 @@ if [ "$1" = "config" ] ; then
   echo "graph_category security"
 fi
 
+if [ "${warning:-}" = "" ] ; then
+  # as default when certificate is only valid for 5 days or less emit a warning
+  warning="5:"
+fi
+if [ "${critical:-}" = "" ] ; then
+  # as default when certificate is only valid for 1 day or less emit a critical
+  critical="1:"
+fi
+
 now=$(date +%s)
+get_validity() {
+  openssl_type=$1
+  file=$2
+  if [ "$file" != "--" ] ; then
+    validity=$(/usr/bin/openssl "$openssl_type" -text -noout -in "$file" | grep -E '(Next Update|Not After)')
+  else
+    # when file is set to -- read from stdin
+    validity=$(/usr/bin/openssl "$openssl_type" -text -noout | grep -E '(Next Update|Not After)')
+  fi
+  validity=${validity#*:}
```

Comment:
I think, I would use multiple variables here. It is usually a rich source of confusion to use the same variable with different content along the flow of the code ...